src/components/map/ha-locations-editor.ts

```diff
@@ -164,6 +164,28 @@ export class HaLocationsEditor extends LitElement {
     }
 
     if (changedProps.has("locations")) {
+      const oldLocations = changedProps.get("locations");
+      const movedLocations = this.locations?.filter(
+        (loc, idx) =>
+          !oldLocations[idx] ||
+          ((loc.latitude !== oldLocations[idx].latitude ||
+            loc.longitude !== oldLocations[idx].longitude) &&
+            this.map.leafletMap?.getBounds().contains({
+              lat: oldLocations[idx].latitude,
+              lng: oldLocations[idx].longitude,
+            }) &&
+            !this.map.leafletMap
+              ?.getBounds()
+              .contains({ lat: loc.latitude, lng: loc.longitude }))
+      );
+      if (movedLocations?.length === 1) {
+        this.updateComplete.then(() =>
```

Comment:
If we need this, why don't we do it in `updated`?

Reply:
No specific reason really, I probably just glommed on to here as we already had related logic for changed locations.

Still sometimes forget or have a hazy understanding of what type of logic is appropriate for which lifecycle methods, esp. when interfacing with more complex components like map.